bionic/flow.py

```diff
@@ -254,22 +302,40 @@ def mark_all_providers_default(self):
 
         return state.touch()
 
-    def _erase_provider(self, name):
+    def _erase_entity_def(self, name):
+        assert name not in self.providers_by_name
+
         state = self
-        if name in state.providers_by_name:
-            state = state.set(providers_by_name=state.providers_by_name.remove(name))
+        state = state.set(entity_defs_by_name=state.entity_defs_by_name.remove(name),)
         if name in state.default_provider_names:
```

Comment:
it feels weird that you are deleting `default_provider_names` inside `_erase_entity_def`. Let's talk about this on monday.

Reply:
You're right, this should now probably be called `default_entity_names`.